### PR TITLE
Fix:EC2-Create-Subnet availability Zone Id support

### PR DESCRIPTION
**Pull request by usmangani1**
Fix:EC2-Create-Subnet availability Zone Id support
https://github.com/localstack/localstack/issues/2788

### CODE REVIEW

File: tests/test_ec2/test_subnets.py [hunk 1]
```diff
@@ -75,6 +75,18 @@ def test_subnet_should_have_proper_availability_zone_set():
...
 
 
+@mock_ec2
+def test_availability_zone_in_create_subnet():
+    ec2 = boto3.resource("ec2", region_name="us-west-1")
+
+    vpc = ec2.create_vpc(CidrBlock="172.31.0.0/16")
+
+    subnet = ec2.create_subnet(
+        VpcId=vpc.id, CidrBlock="172.31.48.0/20", AvailabilityZoneId="use1-az6"
```
Comment (ncarroll0488) [inline tests/test_ec2/test_subnets.py:85]: Should probably change this and #87 to usw1-azX to match the region

**Comment by usmangani1** [comment]
> [Coverage Status](https://coveralls.io/builds/32616932)
> 
> Coverage decreased (-0.1%) to 94.206% when pulling **[706ead9](https://github.com/spulec/moto/commit/706ead9c444c06943abab69a8700f6c0235e1b28) on usmangani1:LISSUE_2788** into **[9894e17](https://github.com/spulec/moto/commit/9894e1785a610f1dd8c9bfce1fa416692b3e0c2f) on spulec:master**.

@bblommers  can you please help me here to get this merged? Both my PR's are pending and the tests are failing at the same test case which is not related to the change.Can you please help me here ?

**Comment by bblommers** [comment]
Thanks @usmangani1 - I've raised #3225 to determine the build failure. 
Will get back to you're PR's after master is fixed.

**Review by bblommers — APPROVED** [review]

**Comment by bblommers** [comment]
This is now part of moto >= 1.3.15.dev1034

**Pull request merged**